### DISCUSSION TITLE
How can I simultaneously play video sounds to OBS?please。

### DISCUSSION THREAD

**Discussion by mrhanCH** [letmaik/pyvirtualcam#104] [Q&A]
I just transmit image data. Can I transmit sound? Thinks

**Comment by longmix** [comment]
Same question confusion me as well .

**Reply by mrhanCH** [reply]
> Same question confusion me as well .
You can use a sub thread to complete audio playback, Then start the sub thread when playing the first frame of the video。
Like this:
PS：audio_devices['input'] and ‘output’ is device index of microphone and audio。


class PlayMusic(QThread):

    wav = None
    def __init__(self):
        super().__init__()
        self.setObjectName('play_music')
    def run(self):
        wf = wave.open(self.wav, 'rb')
        p = pyaudio.PyAudio()
        audio_devices = get_audio_devices()
        stream = p.open(format=p.get_format_from_width(wf.getsampwidth()),
                        channels=wf.getnchannels(),
                        rate=wf.getframerate(),
                        output_device_index=audio_devices['input'],
                        output=True)
        data = wf.readframes(chunk)
        input_stream = p.open(format=FORMAT,
                              channels=CHANNELS,
                              rate=RATE,
                              input=True,
                              input_device_index=audio_devices['output'],
                              frames_per_buffer=chunk)
        while len(data) > 0:
            stream.write(data)
            data = wf.readframes(chunk)
            data2 = input_stream.read(chunk)
        stream.stop_stream()
        stream.close()
        wf.close()
        p.terminate()
